### PR TITLE
[bin2json] Initial commit

### PR DESCRIPTION
**Pull request by paulianttila**
Binary To JSON transformation service

Signed-off-by: Pauli Anttila <pauli.anttila@gmail.com>

### CODE REVIEW

File: bundles/org.openhab.transform.bin2json/src/main/java/org/openhab/transform/bin2json/internal/ConversionException.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+
+    public ConversionException(String message, Throwable cause) {
+        super(message, cause);
+    }
+
+    public ConversionException(Throwable cause) {
+        super(cause);
+    }
+}
```
Comment (davidgraeff) [inline bundles/org.openhab.transform.bin2json/src/main/java/org/openhab/transform/bin2json/internal/ConversionException.java]: newline

File: bundles/org.openhab.transform.bin2json/src/main/java/org/openhab/transform/bin2json/internal/Bin2Json.java [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+                jsn.addProperty(fieldName, ((JBBPFieldUByte) field).getAsInt());
+            } else if (field instanceof JBBPFieldUShort) {
+                jsn.addProperty(fieldName, ((JBBPFieldUShort) field).getAsInt());
+            } else {
+                throw new ConversionException(String.format("Unexpected field '%s'", field));
+            }
+        }
+        return jsn;
+    }
+}
```
Comment (davidgraeff) [inline bundles/org.openhab.transform.bin2json/src/main/java/org/openhab/transform/bin2json/internal/Bin2Json.java]: newline

File: bundles/org.openhab.transform.bin2json/NOTICE [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+of the Eclipse Public License 2.0 which is available at
+https://www.eclipse.org/legal/epl-2.0/.
+
+== Source Code
+
+https://github.com/openhab/openhab2-addons
+
+== Third-party Content
+
+jbbp
```
Comment (davidgraeff) [inline bundles/org.openhab.transform.bin2json/NOTICE]: Can the `jbbp-1.4.1.jar` library be provided by maven as dependency instead of adding the jar file?

**Comment by paulianttila** [comment]
Thanks David. I added it to Karaf features, even it most probably cause merge problems if merge is delayed.

**Comment by paulianttila** [comment]
Forgotten that JBBP jar is not a OSGi bundle so build fails.

**Comment by davidgraeff** [comment]
No you should use the "wrap" prefix. Have a look at all the other bundles in the karaf feature file that are using wrap already.

If there is a maven release, a jar file will no longer be accepted.

**Comment by paulianttila** [comment]
@davidgraeff, that's sounds great...will do.

**Pull request merged**